src/types.cpp

```diff
@@ -0,0 +1,47 @@
+#include "create/types.h"
+
+namespace create {
+  RobotModel::RobotModel(const std::string& name) {
```

Comment:
By removing the public constructor we should only use the three models already defined. For example:

```c++
// Using a ROOMBA_400 model
RobotModel model0 = RobotModel::ROOMBA_400;

// Or a Create 1 model
RobotModel model1 = RobotModel::CREATE_1;

// This is not allowed!
RobotModel model2("CREATE_1");
```